experimental/SetPartitions/src/LinearPartition.jl

```diff
@@ -0,0 +1,201 @@
+        end
+        return new(ring, simplify_operation_zero(coeffs))
+    end
+end 
+
+"""
+    linear_partition(ring::Ring, coeffs::Dict{S, <: Any}) where {S <: AbstractPartition}
+
+Construct a linear partition over `ring` with coefficients `coeffs`.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d") 
+(Univariate polynomial ring in d over QQ, d)
+
+julia> linear_partition(S, Dict(set_partition([1, 2], [1, 1]) => 4, set_partition([1, 1], [1, 1]) => 4*d))
+LinearPartition{SetPartition, QQPolyRingElem}(Univariate polynomial ring in d over QQ, Dict{SetPartition, QQPolyRingElem}(SetPartition([1, 2], [1, 1]) => 4, SetPartition([1, 1], [1, 1]) => 4*d))
+```
+"""
+function linear_partition(ring::Ring, coeffs::Dict{S, <: Any}) where {S <: AbstractPartition}
+    ring_coeffs = Dict(p => ring(c) for (p, c) in coeffs)
+    return LinearPartition{S, elem_type(ring)}(ring, ring_coeffs)
+end
+
+"""
+    linear_partition(ring::Ring, terms::Vector{Tuple{S, <: Any}}) where {S <: AbstractPartition}
+
+Return a `LinearPartition` generated from the Vector `term` of 2-tuples,
+where the first element in the tuple is a `RingElement` and the second
+a `SetPartition`. Furthermore simplify the term before initializing
+the `LinearPartition` object with the corresponding dict.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d")
+(Univariate polynomial ring in d over QQ, d)
+
+linear_partition(S, [(set_partition([1, 1], [1, 1]), 4), (set_partition([1, 1], [1, 1]), 4*d)])
+LinearPartition{SetPartition, QQPolyRingElem}(Univariate polynomial ring in d over QQ, Dict{SetPartition, QQPolyRingElem}(SetPartition([1, 1], [1, 1]) => 4*d + 4))
+```
+"""
+function linear_partition(ring::Ring, terms::Vector{Tuple{S, <: Any}}) where {S <: AbstractPartition}
+    simpl_terms = simplify_operation([(p, ring(c)) for (p, c) in terms])
+    return linear_partition(ring, Dict{S, elem_type(ring)}(simpl_terms))
+end
+
+"""
+    base_ring(p::LinearPartition{S, T}) where {S <: AbstractPartition, T <: RingElem}
+
+Return the underlying coefficient ring of `p`.
+""" 
+function base_ring(p::LinearPartition{S, T}) where {S <: AbstractPartition, T <: RingElement}
+    return p.base_ring::parent_type(T) 
+end
+
+"""
+    coefficients(p::LinearPartition)
+
+Return the coefficients of `p` as dictionary from partitions to elements 
+of the underlying ring.
+""" 
+function coefficients(p::LinearPartition)
+    return p.coefficients
+end
+
+function hash(p::LinearPartition, h::UInt)
+    return hash(base_ring(p), hash(coefficients(p), h)) 
+end
+
+function ==(p::LinearPartition, q::LinearPartition)
+    return base_ring(p) == base_ring(q) && coefficients(p) == coefficients(q)
+end
+
+function deepcopy_internal(p::LinearPartition, stackdict::IdDict)
+    if haskey(stackdict, p)
+        return stackdict[p]
+    end
+    q = linear_partition(
+            deepcopy_internal(base_ring(p), stackdict),
```

Comment:
```suggestion
            base_ring(p),
```
In most cases you don't want to copy the base ring as well when copying a partition. If you do this, then e.g. `x + deepcopy(x)` won't work (e.g. You cannot use deep copy when defining arithmetics)